extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/internal/discovery/HueLightDiscoveryService.java

```diff
@@ -144,13 +144,15 @@ private void onLightAddedInternal(FullLight light) {
         ThingUID thingUID = getThingUID(light);
         ThingTypeUID thingTypeUID = getThingTypeUID(light);
 
-        String modelId = light.getModelID().replaceAll(HueLightHandler.NORMALIZE_ID_REGEX, "_");
+        String modelId = light.getModelID();
 
         if (thingUID != null && thingTypeUID != null) {
             ThingUID bridgeUID = hueBridgeHandler.getThing().getUID();
             Map<String, Object> properties = new HashMap<>();
             properties.put(LIGHT_ID, light.getId());
-            properties.put(Thing.PROPERTY_MODEL_ID, modelId);
+            if (modelId != null) {
```

Comment:
If `modelId` is only used if `thingUID != null && thingTypeUID != null` I would move the variable definition and initialization into this branch.

Reply:
No, it is used in the `else` branch too.